packages/nadle/src/core/reporter.ts

```diff
@@ -70,7 +70,8 @@ export class DefaultReporter implements Reporter {
 	}
 
 	private printRunningTasks() {
-		const lines = Array.from({ length: this.nadle.options.maxWorkers }, () => ` ${c.yellow(">")} ${c.dim("IDLE")}`);
+		const runningWorkers = new Set(Object.values(this.threadIdPerWorker)).size;
+		const lines = Array.from({ length: runningWorkers }, () => ` ${c.yellow(">")} ${c.dim("IDLE")}`);
 
 		for (const runningTask of this.nadle.registry.getAll().filter((task) => task.status === TaskStatus.Running)) {
 			lines[this.threadIdPerWorker[runningTask.name] - 1] = ` ${c.yellow(">")} :${c.bold(runningTask.name)}`;
```

Comment:
What if the `threadIdPerWorker = {A: 1, B: 2, C: 5}` while the `lines` has length 3?

Reply:
I updated it myself. Hope it still fine 😳